src/platform/api.ts

```diff
@@ -87,20 +87,29 @@ export function buildApi(
             await service.createNew(options);
         },
         getKernelService: async () => {
-            const kernelServiceFactory = serviceContainer.get<JupyterKernelServiceFactory>(JupyterKernelServiceFactory);
+            const kernelServiceFactory =
+                serviceContainer.get<IExportedKernelServiceFactory>(IExportedKernelServiceFactory);
             return kernelServiceFactory.getService();
         }
     };
 
     // In test/dev environment return the DI Container.
     if (
         isTestExecution() ||
-        process.env.VSC_JUPYTER_EXPOSE_SVC ||
-        context.extensionMode === ExtensionMode.Development
+        context.extensionMode === ExtensionMode.Development ||
+        context.extensionMode === ExtensionMode.Test
     ) {
         /* eslint-disable @typescript-eslint/no-explicit-any */
         (api as any).serviceContainer = serviceContainer;
         (api as any).serviceManager = serviceManager;
+        (api as any).getSymbol = (symbol: String) => {
+            // Needed for web tests. Eval for the extension happens in another web worker so
```

Comment:
When running the web tests, the `Symbol(IApplicationShell)` was not the same value as found in the container. So I added this extra export to allow a different bundle to get the correct value of symbols.

I believe what's happening is that the extension bundle is eval'd and then in a separate worker, the test bundle is eval'd. Global data is different between them, so I needed a way to export it from the extension bundle.

This function will need to be updated as we write more tests for the web extension.